aiida/backends/managers/access.py

```diff
@@ -0,0 +1,259 @@
+            (enabled by default).
+        """
+        return self.get_processes_recorded_in(self._path_to_tracked, ignore_self=ignore_self)
+
+    def acquire_lock(self):
+        """Creates a locking file for the running process"""
+        tracked_processes = self.get_tracked_processes()
+        filtered_processes = self.separate_processes(tracked_processes)
+        self.delete_process_records(filtered_processes['terminated'])
+
+        if len(filtered_processes['remaining']) > 0:
+            raise exceptions.LockingProfileError('can`t lock profile')
+
+        process_data = ProcessData()
+        process_idn = process_data.pid
+        filepath_temp = self._path_to_locking / f'request_{process_idn}.pid'
+        filepath_final = self._path_to_locking / f'{process_idn}.pid'
+        process_data.write_to_file(filepath_temp)
+        os.rename(filepath_temp, filepath_final)
+
+        locking_processes = self.get_locking_processes()
+        if len(locking_processes) > 0:
+            os.remove(filepath_final)
+            raise exceptions.LockedProfileError('can`t lock profile')
+
+        tracked_processes = self.get_tracked_processes()
+        if len(tracked_processes) > 0:
+            os.remove(filepath_final)
+            raise exceptions.LockedProfileError('can`t lock profile')
+
+    def release_lock(self):
+        """Releases the locking file of the running process"""
+        process_data = ProcessData()
+        process_idn = process_data.pid
+        filepath = self._path_to_locking / f'{process_idn}.pid'
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(filepath)
+
+    @contextlib.contextmanager
+    def profile_locking_context(self):
+        """Context for working with a locked profile"""
+        try:
+            self.acquire_lock()
+            yield
+        finally:
+            self.release_lock()
+
+    def record_process_access(self, verify_lock=True):
+        """Records the access of a process
+
+        :param verify_lock:
+            flag to check if there is a lock on the profile that would prevent access (True by default,
+            otherwise it will ignore the lock and provide access anyway). The check is performed both
+            at the start and the end of the process to check for any posible racing conditions.
+        """
+        if verify_lock:
+            locking_processes = self.get_locking_processes()
+            if len(locking_processes) > 0:
+                raise exceptions.LockingProfileError('can`t access profile')
+
+        process_data = ProcessData()
+        process_idn = process_data.pid
+        filepath_temp = self._path_to_tracked / f'request_{process_idn}.pid'
+        filepath_final = self._path_to_tracked / f'{process_idn}.pid'
+        process_data.write_to_file(filepath_temp)
+        os.rename(filepath_temp, filepath_final)
+
+        if verify_lock:
```

Comment:
Here is indeed the place where I was mentioning that the responsibility should be for not starting. So this code is very important. Let's add a clear comment explaining what is this part of the code doing and why it's important.